### PR TITLE
Add dof_exemption datasets

### PR DESCRIPTION
**Pull request by sraby**
This PR adds a new dataset from the NYC Open Data portal that help us track all different kinds of property tax exemptions and abatements—the [Property Exemption Detail](https://data.cityofnewyork.us/City-Government/Property-Exemption-Detail/muvi-b6kx), along with its corresponding [Exemption Classification Codes](https://data.cityofnewyork.us/City-Government/Exemption-Classification-Codes/myn9-hwsy).

### CODE REVIEW

**Comment by sraby** [comment]
@lblok @austensen @aepyornis here's another one for y'all!

File: src/tests/integration/test_nycdb.py [hunk 1]
```diff
@@ -125,6 +125,22 @@ def test_hpd_complaints(conn):
     assert row_count(conn, 'hpd_complaints') == 100
 
 
+def test_dof_exemptions(conn):
+    drop_table(conn, 'dof_exemptions')
+    drop_table(conn, 'dof_exemption_classification_codes')
+    hpd_complaints = nycdb.Dataset('dof_exemptions', args=ARGS)
```
Comment (samaratrilling) [inline src/tests/integration/test_nycdb.py:131]:
```suggestion
    dof_exemptions = nycdb.Dataset('dof_exemptions', args=ARGS)
```
Maybe name this according to what the data is, like we've done below

File: src/tests/integration/test_nycdb.py [hunk 2]
```diff
@@ -125,6 +125,22 @@ def test_hpd_complaints(conn):
...
+    hpd_complaints.db_import()
+    assert row_count(conn, 'dof_exemptions') == 10
+
+
+def test_dof_exemption_classification_codes(conn):
+    drop_table(conn, 'dof_exemptions')
+    drop_table(conn, 'dof_exemption_classification_codes')
+    hpd_complaints = nycdb.Dataset('dof_exemptions', args=ARGS)
```
Comment (samaratrilling) [inline src/tests/integration/test_nycdb.py:139]:
```suggestion
    dof_exemption_classification_codes = nycdb.Dataset('dof_exemption_classification_codes', args=ARGS)
```
Should use the classification codes dataset here to make sure we're testing both datasets.

**Review by samaratrilling — APPROVED** [review]
Approved once the dataset name is switched in that test and the variables are renamed for clarity!

File: src/tests/integration/test_nycdb.py (hunk 1 above)
Comment (sraby) [reply src/tests/integration/test_nycdb.py:131]: GAH thank you! I was using the hpd_complaints dataset as a template and inevitably I forgot to change a few variables...

File: src/tests/integration/test_nycdb.py (hunk 2 above)
Comment (sraby) [reply src/tests/integration/test_nycdb.py]: So, my understanding is we actually want to keep the name of the dataset `dof_exemptions` as that's the key name of the larger "dataset" in nycdb that includes the two tables. I get this error when I try and chang the name to `dof_exemption_classification_codes`, which speaks to this issue: 
```
self = <nycdb.dataset.Dataset object at 0x7f6616f0b850>
dataset_name = 'dof_exemption_classification_codes'
args = namespace(user='nycdb', password='nycdb', host='db', database='nycdb_test', port='5432', root_dir='/nycdb/src/tests/integration/data', hide_progress=False)

    def __init__(self, dataset_name, args=None):
        self.name = dataset_name
        self.args = args
        self.db = None
    
        if self.args:
            self.root_dir = self.args.root_dir
        else:
            self.root_dir = './data'
    
>       self.dataset = datasets()[dataset_name]
E       KeyError: 'dof_exemption_classification_codes'

nycdb/dataset.py:46: KeyError
```

Comment (sraby) [reply src/tests/integration/test_nycdb.py]: That being said, I think we are still able to test the `dof_exemption_classification_codes` table in the `assert clause, given that both tables in the dataset get imported by the `.db_import()` command:
```
dof_exemptions = nycdb.Dataset('dof_exemptions', args=ARGS)
dof_exemptions.db_import()
assert row_count(conn, 'dof_exemption_classification_codes') == 10
   ```

Comment (samaratrilling) [reply src/tests/integration/test_nycdb.py]: Ahh, understood. Looks fine then! Thanks!

**Pull request merged**